File: Reverse Linked List.py

# https://leetcode.com/problems/reverse-linked-list/
# 24.10.2020


class Solution(object):
    def reverseList(self, head):

        if head == None:
            return head

        temp = head
        arr = []
        while temp.next:
            arr.append(temp.val)
            temp = temp.next
        arr.append(temp.val)

        temp = head
        temp.val = arr.pop()
        for i in range(len(arr)):
            temp = temp.next
            temp.val = arr.pop()

        return head

File: reverse-linked-list.cs

# https://leetcode.com/problems/reverse-linked-list/
# 27.03.2024

/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution
{
    public ListNode ReverseList(ListNode head)
    {
        ListNode prev = null;
        ListNode current = head;
        ListNode next;

        while (current != null)
        {
            next = current.next;
            current.next = prev;
            prev = current;
            current = next;
        }

        return prev;
    }
}
